src/compiler/msvc.rs

```diff
@@ -195,44 +195,46 @@ fn encode_path(dst: &mut Write, path: &Path) -> io::Result<()> {
     dst.write_all(&bytes)
 }
 
-#[derive(Clone, Debug)]
-enum MSVCArgAttribute {
-    TooHard,
-    PreprocessorArgument,
-    DoCompilation,
-    ShowIncludes,
-    Output,
-    DepFile,
-    ProgramDatabase,
-    DebugInfo,
-    XClang,
+ArgData!{
+    TooHardFlag(()),
+    TooHard(OsString),
+    TooHardPath(PathBuf),
```

Comment:
Since we don't really need the argument for `TooHardPath`, can we just drop this and use `TooHard(OsString)` for those arguments?

Further musing has me wonder if we could combine them all into `TooHard(Option<OsString>)` and just impl the right traits for `Option<OsString>`.